### PR TITLE
added subsystem class

### PR DESCRIPTION
**Pull request by Bar2005**
waiting for your review

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 1]
```diff
@@ -1,53 +1,104 @@
...
-        public static RobotState getRobotState ( final RobotState userWantedState){
-            return getStateFromWantedAndCurrent(GlobalData.driverPressedButtons, userWantedState);
-        }
 
-    private static RobotState prevRobotState = GlobalData.robotState;
 
-    public static void setSubsystemToState(final RobotState robotState) {
-        switch (robotState) {
+    private static void setSubsystemToState(Gamepad gamepad1, Gamepad gamepad2) {
+        elevatorState = getElevatorStateFromSecondDriver(gamepad2);
```
Comment (Ido-Sobol) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:47]: Need to do here the same null safety logics we did with the first gamepad

**Review by Ido-Sobol — CHANGES_REQUESTED** [review]

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 2]
```diff
@@ -1,53 +1,104 @@
...
-        }
-
-        // // Set the state of the LED for telemetry.
-        // LED.setState();
 
-        prevRobotState = robotState;
+        }
+        if (gamepad1.right_bumper){
+            armState = armState == ArmState.FRONT ? ArmState.BACK : ArmState.FRONT;
+        }
```
[the remark this answers is not in the document]
Comment (Bar2005) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:97]: But we didnt write any button that should perform this

[the remark this answers is not in the document]
Comment (Ido-Sobol) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:97]: Y is the button for it - look in the scheme

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 3]
```diff
@@ -1,53 +1,104 @@
...
+    private static void setSubsystemToState(Gamepad gamepad1, Gamepad gamepad2) {
+        elevatorState = getElevatorStateFromSecondDriver(gamepad2);
+        switch (state) {
             case TRAVEL:
-            default:
-                // Intake.set(IntakeState.CLOSED);
+                intakeState = IntakeState.STOP;
+                if (isGamePiece) {
+                    clawState = ClawState.CLOSE;
+                    if (clawServo.getPosition() == ClawConstants.closed) {
```
[the remark this answers is not in the document]
Comment (Ido-Sobol) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:53]: @water221c on it?

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java [hunk 4]
```diff
@@ -1,53 +1,104 @@
...
-        // : (buttons[HardwareMap.psXIndex]) ? RobotState.INTAKE
-        // : (buttons[HardwareMap.psOptionsIndex] || GlobalData.isPOVPressed) ?
-        // RobotState.CLIMB
-        // : GlobalData.robotState;
-        return RobotState.TRAVEL;
+    public static RobotState state = RobotState.TRAVEL;
+    static ElevatorStates elevatorState = ElevatorStates.INTAKE;
+    static ClawState clawState = ClawState.CLOSE;
+    static ArmState armState = ArmState.BACK;
+    static IntakeState intakeState = IntakeState.STOP;
```
[the remark this answers is not in the document]
Comment (water221c) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:26]: done

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java (hunk 1 above)
Comment (water221c) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:47]: done

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (water221c) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:53]: done

[the remark this answers is not in the document]
Comment (water221c) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robotSubSystems/SubSystemManager.java:53]: I took care of all the reviews, pushing now and waiting for more notes

**Comment by Ido-Sobol** [comment]
Nicely done!

**Pull request merged**